SingleMeshBlend/SingleMeshBlend/SingleBlendMeshDeformer.cpp

```diff
@@ -204,37 +204,44 @@ MThreadRetVal SingleBlendMeshDeformer::threadEvaluate(void * pParam)
 	unsigned int start{ threadData->start };
 	unsigned int end{ threadData->end };
 
-	MPointArray& vertexPositions{ data->vertexPositions };
-	MPointArray& blendVertexPositions{ data->blendVertexPositions };
+	//Operator [] on maya containers as  a non negligible overhead. By using pointer we can bypass it
+	MPoint* currentVertexPosition{ &data->vertexPositions[0] };
+	MVectorArray deltas{ data->deltas };
 
 	float envelopeValue{ data->envelopeValue };
 	double blendWeightValue{ data->blendWeightValue };
 
-	//MPointArray operator[] has a non-negligible overhead. 
-	//Accessing the memory directly with pointers bypass that overhead.
-	unsigned int vertexCount{ vertexPositions.length() };
-	MPoint* currentVertexPosition{ &vertexPositions[0] };
-	MPoint* blendVertexPosition{ &blendVertexPositions[0] };
-
-	for (unsigned int vertexIndex{ start }; vertexIndex < end; vertexIndex++) {
-		MVector delta{ (*(blendVertexPosition + vertexIndex) - *(currentVertexPosition + vertexIndex)) * blendWeightValue * envelopeValue };
-		MPoint newPosition{ delta + *(currentVertexPosition + vertexIndex) };
-
-		*(currentVertexPosition + vertexIndex) = newPosition;
+	unsigned int vertexCount{ data->vertexPositions.length() };
+	for (unsigned int vertexIndex{ start }; vertexIndex < end; ++vertexIndex) {
+		currentVertexPosition[vertexIndex] = (deltas[vertexIndex] * blendWeightValue * envelopeValue) + currentVertexPosition[vertexIndex];
```

Comment:
blendWeightValue* envelopeValue is constant, you can compute that before the loop and use the same value. Probably the compiler is optimizing that by its own but good practice to do that anyway.

Reply:
Interesting. Did not think about it. will do.